course5/homework2/carmack/carmack.cc

```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include <math.h>
+#include <time.h>
+#include <string>
+
+// Implementing fast-sqrt algortihm. However, for me, sqrt from the math library seems much faster.
+
+float inv_sqrt( float x )
+{
+    int i;
```

Comment:
It is essential that the integer is 32 bit, therefore it is better to use one of the explicitly-sized typedefs (```std::int32_t```)